src/main/java/gov/cms/mat/cql_elm_translation/cql_translator/MadieLibrarySourceProvider.java

```diff
@@ -11,17 +11,39 @@
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 @Slf4j
 public class MadieLibrarySourceProvider implements LibrarySourceProvider {
+
+  private static final String[] STRING_ARR = new String[0];
   private static final ConcurrentHashMap<String, String> cqlLibraries = new ConcurrentHashMap<>();
   private static final ThreadLocal<UsingProperties> threadLocalValue = new ThreadLocal<>();
   private static final ThreadLocal<String> threadLocalValueAccessToken = new ThreadLocal<>();
-  private static MadieFhirServices madieFhirServices;
+  private static CqlLibraryService cqlLibraryService;
+  private static final Map<String, String[]> supportedLibrariesMap = Map.of(
+    "FHIR", List.of("FHIR").toArray(STRING_ARR),
+    "QICORE", List.of("FHIR", "QICore").toArray(STRING_ARR),
+    "QDM", List.of("QDM").toArray(STRING_ARR)
+  );
+
+  public static String getAccessToken() {
+    return threadLocalValueAccessToken.get();
+  }
+
+  public static UsingProperties getUsingProperties() {
+    return UsingProperties.builder()
+        .libraryType(threadLocalValue.get().getLibraryType())
+        .version(threadLocalValue.get().getVersion())
+        .line(threadLocalValue.get().getLine())
+        .comment(threadLocalValue.get().getComment())
+        .build();
+  }
 
-  public static void setFhirServicesService(MadieFhirServices madieFhirServices) {
-    MadieLibrarySourceProvider.madieFhirServices = madieFhirServices;
+  public static void setFhirServicesService(CqlLibraryService cqlLibraryService) {
```

Comment:
Nothing major, but Is there a better name for this method, it's trippy and FHIR could be omitted. Maybe setSourceProvider or setSourceProviderService

Reply:
oh - totally right! I missed this when I was using the IDE refactor -> rename functions!